### ISSUE TITLE
Odd `<mark>...</mark>` ranges

### ISSUE THREAD

**Issue by kdheepak** [jameslittle230/stork#292]
I'm using the latest `stork`

```
$ stork --version
Stork 1.4.2
```

This is an example of the snippet in the search result when the search query is "neovim":

```
<p>
        ...to use this you will need the following: <mark class="stork-highlight">neovim</mark> v0.5.0 <mark class="stork-highlight">`neovim/nvim-ls</mark>p` The <mark class="stork-highlight">neovim/nvim-lsp</mark> repository contains language server configurations for a...
        </p>
```

You can see that when backticks are involved, the `mark` seems to exclude the last letter of the search result:

<img width="930" alt="Screen Shot 2022-05-01 at 11 30 22 PM" src="https://user-images.githubusercontent.com/1813121/166188562-dc756631-12b2-433e-913e-3d609f22949f.png">

The end result is some rather old highlighting. This is what it looks like when I'm not showing the console highlighting:

<img width="700" alt="Screen Shot 2022-05-01 at 11 30 56 PM" src="https://user-images.githubusercontent.com/1813121/166188619-026c2d50-a107-4ff1-b615-913113a24f15.png">

For reproducing this, you can find a link to the `.st` file here:

https://github.com/kdheepak/blog/blob/9e8c45bc90395d05cc610c82f44d6926d4eda7f4/static/assets/stork/search.st

You can also download it here:

https://blog.kdheepak.com/assets/stork/search.st

The code I'm using is all standard, based on the documentation: https://github.com/kdheepak/blog/blob/9e8c45bc90395d05cc610c82f44d6926d4eda7f4/src/lib/components/Search.svelte

And I'm loading stork from the CDN as shown in the documentation:

https://github.com/kdheepak/blog/blob/9e8c45bc90395d05cc610c82f44d6926d4eda7f4/src/app.html#L7

You should be able to see a live version of this bug in action here: https://blog.kdheepak.com/. Scroll to the bottom and you'll be able to find the search bar.

**Comment by jameslittle230** [comment]
Thanks for the detailed report! I'll look into this and see if I can make improvements here. As far as I know, this bug is coming into play because there's a hyphen in the word being searched, and some parts of Stork treat a hyphen as a word separator (like a space) whereas others do not.

**Comment by kdheepak** [comment]
It's possible it is related to hypens, but it also occurs elsewhere. For example, if I search for `vimrc`:

<img width="1187" alt="Screen Shot 2022-05-04 at 7 06 56 PM" src="https://user-images.githubusercontent.com/1813121/166850400-f4797733-ef1d-45f9-be67-ba675c66bb67.png">

It always occurs for text in `<mark></mark>` tags.

**Comment by jameslittle230** [comment]
Ah - might be more than just hyphens, might be all punctuation.

I'll take a look.

**Comment by kdheepak** [comment]
Thanks! Great job on stork btw!

**Comment by jameslittle230** [comment]
This has been fixed in #297! It'll be released in version 1.5.0, which will be released as soon as I get a moment where I have time to do so!

**Comment by kdheepak** [comment]
Awesome! No hurry on my end!
